applications/GeoMechanicsApplication/tests/cpp_tests/test_geomechanics_time_integration_scheme.cpp

```diff
@@ -0,0 +1,220 @@
+            mScheme.FinalizeSolutionStep(GetModelPart(), A, Dx, b);
+        };
+
+        auto initialize_function = [this, &A, &Dx, &b]() {
+            mScheme.InitializeSolutionStep(GetModelPart(), A, Dx, b);
+        };
+        auto initialize_non_linear_iteration = [this, &A, &Dx, &b]() {
+            mScheme.InitializeNonLinIteration(GetModelPart(), A, Dx, b);
+        };
+
+        auto finalize_non_linear_iteration = [this, &A, &Dx, &b]() {
+            mScheme.FinalizeNonLinIteration(GetModelPart(), A, Dx, b);
+        };
+
+        // Create functions that check if the previously mentioned functions have been called
+        auto finalize_function_check = [](const Kratos::intrusive_ptr<T> rElement) {
+            return rElement->IsSolutionStepFinalized();
+        };
+
+        auto initialize_function_check = [](const Kratos::intrusive_ptr<T> rElement) {
+            return rElement->IsSolutionStepInitialized();
+        };
+
+        auto initialize_non_linear_iteration_check =
+            [](const Kratos::intrusive_ptr<T> rCondition) {
+                return rCondition->IsNonLinIterationInitialized();
+            };
+
+        auto finalize_non_linear_iteration_check = [](const Kratos::intrusive_ptr<T> rCondition) {
+            return rCondition->IsNonLinIterationFinalized();
+        };
+
+        functions_and_checks.push_back({finalize_solution_step, finalize_function_check});
+        functions_and_checks.push_back({initialize_function, initialize_function_check});
+        functions_and_checks.push_back({initialize_non_linear_iteration,
+                                        initialize_non_linear_iteration_check});
+        functions_and_checks.push_back({finalize_non_linear_iteration,
+                                        finalize_non_linear_iteration_check});
+
+        return functions_and_checks;
+    }
+
+    template <class T>
+    std::vector<std::pair<std::function<void(const Kratos::intrusive_ptr<T> Component)>, std::function<bool(const Kratos::intrusive_ptr<T> rElement)>>> CreateFunctionsAndChecksCalledOnASingleComponent()
+    {
+        std::vector<std::pair<std::function<void(const Kratos::intrusive_ptr<T> Component)>,
+                              std::function<bool(const Kratos::intrusive_ptr<T> Component)>>>
+            functions_and_checks;
+
+        Element::EquationIdVectorType equation_id_vector_type;
+        ProcessInfo info;
+
+        auto equation_id = [this, &equation_id_vector_type,
+                            &info](const Kratos::intrusive_ptr<T> Component) {
+            mScheme.EquationId(*Component.get(), equation_id_vector_type, info);
+        };
+
+        auto equation_id_check = [](const Kratos::intrusive_ptr<T> Component) {
+            return Component->IsEquationIdSet();
+        };
+
+        Element::DofsVectorType dofs;
+        auto get_dofs_list = [this, &dofs, &info](const Kratos::intrusive_ptr<T> Component) {
```

Comment:
Also here I believe we will have a dangling reference. You may fix it in the same way as previously suggested.